File: packages/core/ios/Logger.swift
import os

public struct Logger {
    // Replace with your own bundle identifier or subsystem
    private static let subsystem = Bundle.main.bundleIdentifier ?? "com.your.app"
    private static let log = os.Logger(subsystem: subsystem, category: "ReactNative AMA")

    public static func info(_ fn: String, _ message: String, extra: String? = nil) {
        if let extra = extra, !extra.isEmpty {
            log.info("[INFO]: \(fn) \(message)  >>> \(extra, privacy: .public)")
        } else {
            log.info("[INFO]: \(fn) \(message, privacy: .public)")
        }
    }

    public static func info2(_ fn: String, _ message: String) {
        info("  \(fn)", message)
    }

    public static func debug(_ fn: String, _ message: String, extra: String? = nil) {
        if let extra = extra, !extra.isEmpty {
            log.debug("[DEBUG]: \(fn) \(message)  >>> \(extra, privacy: .public)")
        } else {
            log.debug("[DEBUG]: \(fn) \(message, privacy: .public)")
        }
    }

    public static func error(_ fn: String, _ message: String) {
        log.error("[ERROR]: \(fn) \(message, privacy: .public)")
    }
}

File: packages/core/android/src/debug/java/expo/modules/ama/logger.kt
package expo.modules.ama

import android.util.Log

object Logger {
    public fun info(fn: String, message: String, extra: String? = null) {
        if (extra.isNullOrEmpty()) {
            Log.i("[ReactNative AMA]: ", fn + " " + message)
        } else {
            Log.i("[ReactNative AMA]: ", fn + " " + message + "  >>> " + extra)
        }
    }

    public fun info2(fn: String, message: String) {
        Logger.info("  " + fn, message)
    }

    public fun debug(fn: String, message: String) {
        // Log.d("[ReactNative AMA]: ", fn + " " + message)
    }

    public fun error(fn: String, message: String) {
        Log.e("[ReactNative AMA - Error]: ", fn + " " + message)
    }
}
